library/Kubernetes/Web/ServiceDetail.php

```diff
@@ -91,15 +92,22 @@ protected function assemble(): void
                     Filter::equal('pod.label.value', $selector->value)
                 ));
             }
-            $this->addHtml((new PodList(
-                $pods
-            ))->setWrapper(new HtmlElement(
-                'section',
-                null,
-                new HtmlElement('h2', null, new Text($this->translate('Pods')))
-            )));
+
+            if (Auth::getInstance()->hasPermission(Auth::SHOW_PODS)) {
```

Comment:
Move that to the `if ($selectors->valid())` above. Then apply the restrictions right before the `foreach ($selectors ...)`.